tests/Unit/ControlSystem/Test_PiecewisePolynomial.cpp

```diff
@@ -0,0 +1,141 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "ControlSystem/PiecewisePolynomial.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+SPECTRE_TEST_CASE("Unit.ControlSystem.FunctionsOfTime.PiecewisePolynomial",
+                  "[ControlSystem][Unit]") {
+  double t = 0.0;
+  const double dt = 0.6;
+  const double final_time = 4.0;
+  const constexpr size_t deriv_order = 3;
+
+  // test two component system (x**3 and x**2)
+  const std::array<DataVector, deriv_order + 1> init_func{
+      {{0.0, 0.0}, {0.0, 0.0}, {0.0, 2.0}, {6.0, 0.0}}};
+  FunctionsOfTime::PiecewisePolynomial<deriv_order> f_of_t(t, init_func);
+
+  while (t < final_time) {
+    const auto& lambdas = f_of_t(t);
+
+    CHECK(approx(lambdas[0][0]) == cube(t));
+    CHECK(approx(lambdas[0][1]) == square(t));
+
+    t += dt;
+    f_of_t.update(t, {6.0, 0.0});
+  }
+  // test time_bounds function
+  const auto& t_bounds = f_of_t.time_bounds();
+  CHECK(t_bounds[0] == 0.0);
+  CHECK(t_bounds[1] == 4.2);
+}
+
+SPECTRE_TEST_CASE(
+    "Unit.ControlSystem.FunctionsOfTime.PiecewisePolynomial.FirstDerivs",
+    "[ControlSystem][Unit]") {
+  double t = 0.0;
+  const double dt = 0.6;
+  const double final_time = 4.0;
+  const constexpr size_t deriv_order = 3;
+  const constexpr size_t ret_deriv_order = 1;
+
+  // test two component system (x**3 and x**2)
+  const std::array<DataVector, deriv_order + 1> init_func{
+      {{0.0, 0.0}, {0.0, 0.0}, {0.0, 2.0}, {6.0, 0.0}}};
+  FunctionsOfTime::PiecewisePolynomial<deriv_order> f_of_t(t, init_func);
+
+  while (t < final_time) {
+    const auto& lambdas = f_of_t.operator()<ret_deriv_order>(t);
```

Comment:
Combine this test with the previous, adding this line and the CHECKs inside the loop above